File: Data Structures & Algorithms/longest-consecutive-sequence/submission-3.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if not nums:
            return 0
        nums.sort()
        res = 0

        lcs = 1
        curr = 1

        for i in range(1, len(nums)):
            if nums[i] == nums[i - 1]:
                continue
            elif nums[i] == nums[i-1] + 1:
                curr += 1
            else:
                lcs = max(lcs, curr)
        res = max(lcs, curr)
        return res
        

File: Data Structures & Algorithms/binary-search/submission-3.cpp
class Solution {
public:
    int binary_search(int l, int r, vector<int>& nums, int target){
        if(l <= r){
            int m = l + (r - l) / 2;
            if(nums[m] == target) return m;
            if(nums[m] > target){
                return binary_search(l, m - 1, nums, target);
            }
            else{
                return binary_search(m + 1, r, nums, target);
            }
        }
        return -1;
    }

    int search(vector<int>& nums, int target) {
        return binary_search(0, nums.size() - 1, nums, target);
    }
};
